src/map_io/map_building_packet.cc

```diff
@@ -165,24 +163,26 @@ void MapBuildingPacket::write(FileSystem& fs, EditorGameBase& egbase, MapObjectS
 }
 
 // TODO(Nordfriese): Savegame compatibility. Delete this function after v1.0
-void MapBuildingPacket::read_priorities(Building& building, FileRead& fr) {
+void MapBuildingPacket::read_priorities(EditorGameBase& egbase, Building& building, FileRead& fr) {
 	// TODO(unknown): savegame compatibility
 	fr.unsigned_32();  // unused, was base_priority which is unused. Remove after b20.
 
-	const TribeDescr& tribe = building.owner().tribe();
-	Widelands::DescriptionIndex ware_type;
-	// read ware type
+	Descriptions* descriptions = egbase.mutable_descriptions();
+
+	// read ware/worker type
+	Widelands::DescriptionIndex ware_type = INVALID_INDEX;
 	while (0xff != (ware_type = fr.unsigned_8())) {
 		// read count of priorities assigned for this ware type
 		const uint8_t count = fr.unsigned_8();
 		for (uint8_t i = 0; i < count; ++i) {
-			DescriptionIndex idx;
-			if (wwWARE == ware_type) {
-				idx = tribe.safe_ware_index(fr.c_string());
-			} else if (wwWORKER == ware_type) {
-				idx = tribe.safe_worker_index(fr.c_string());
-			} else {
-				throw GameDataError("unrecognized ware type %d while reading priorities", ware_type);
+			DescriptionIndex idx = Widelands::INVALID_INDEX;
+			switch (ware_type) {
+			case WareWorker::wwWARE:
+				idx = descriptions->load_ware(fr.c_string());
+				break;
+			case WareWorker::wwWORKER:
+				idx = descriptions->load_worker(fr.c_string());
+				break;
 			}
```

Comment:
Why not?
   default:
      throw GameDataError("unrecognized ware type %d while reading priorities", ware_type);